tests/cephfs/lib/cephfs_common_lib.py

```diff
@@ -250,3 +256,175 @@ def test_cleanup(self, client, setup_params, mount_details):
+                ).strip(),
+            )
+            log.info(f"Sub volume path: {subvol_path.strip()}")
+
+        if mount_type == "fuse":
+            if subvolume_name:
+                self.fuse_mount(
+                    [client],
+                    mount_path,
+                    extra_params=f" --client_fs {fs_name} -r {subvol_path.strip()}",
+                )
+            else:
+                self.fuse_mount(
+                    [client], mount_path, extra_params=f" --client_fs {fs_name}"
+                )
+
+        elif mount_type == "smb":
+            smb_subvol_group = kwargs.get("smb_subvolume_group", "smb")
+            smb_subvols = kwargs.get("smb_subvolumes", ["sv1"])
+            smb_subvolume_mode = kwargs.get("smb_subvolume_mode", "0777")
+            smb_cluster_id = kwargs.get("smb_cluster_id", "smb1")
+            auth_mode = kwargs.get("auth_mode", "user")
+            domain_realm = kwargs.get("domain_realm", None)
+            custom_dns = kwargs.get("custom_dns", None)
+            smb_user_name = kwargs.get("smb_user_name", "user1")
+            smb_user_password = kwargs.get("smb_user_password", "passwd")
+            smb_shares = kwargs.get("smb_shares", ["share1", "share2"])
+            path = kwargs.get("path", "/")
+            installer = self.ceph_cluster.get_nodes(role="installer")[0]
+            smb_nodes = self.ceph_cluster.get_nodes("smb")
+            client = self.ceph_cluster.get_nodes(role="client")[0]
+            mount_path = kwargs.get("cifs_mount_point", "/mnt/smb")
+
+            try:
+                # deploy smb services
+                deploy_smb_service_imperative(
+                    installer,
+                    fs_name,
+                    smb_subvol_group,
+                    smb_subvols,
+                    smb_subvolume_mode,
+                    smb_cluster_id,
+                    auth_mode,
+                    smb_user_name,
+                    smb_user_password,
+                    smb_shares,
+                    path,
+                    domain_realm,
+                    custom_dns,
+                )
+
+                # Check smb share using smbclient
+                smbclient_check_shares(
+                    smb_nodes,
+                    client,
+                    smb_shares,
+                    smb_user_name,
+                    smb_user_password,
+                    auth_mode,
+                    domain_realm,
+                )
+
+                # Mount smb share with cifs
+                smb_cifs_mount(
+                    smb_nodes[0],
+                    client,
+                    smb_shares[0],
+                    smb_user_name,
+                    smb_user_password,
+                    auth_mode,
+                    domain_realm,
+                    mount_path,
+                )
+
+            except Exception as e:
+                log.error("failed to setup smb with error: {}".format(e))
+                return 1
+
```

Comment:
IMHO, we could write separate methods to handle this chunk of code. It is unreadable when you keep adding `conditions`.

My rule of thumb is to avoid `else` statements as much as possible.